app/models/round.rb

```diff
@@ -121,6 +121,42 @@ def name
     Round.name_from_attributes(event, round_type)
   end
 
+  def registrations
+    if number == 1
+      Registration.joins(:registration_competition_events)
+                  .where(
+                    competition_id: competition_event.competition_id,
+                    competing_status: 'accepted',
+                    registration_competition_events: { competition_event_id: competition_event_id }
+                  ).includes([:user])
+    else
+      previous_round = Round.joins(:competition_event).find_by(competition_event: { competition_id: competition_event.competition_id, event_id: event.id }, number: number - 1)
+      previous_round.live_results.where(advancing: true).includes(:registration).map(&:registration)
```

Comment:
Ooof, that's a lot of database load. Can we somehow pluck registration IDs with a single `SELECT` and then load registrations based on that?

Reply:
yeah, I was thinking something like that too